lib/archethic/contracts/worker.ex

```diff
@@ -256,23 +256,37 @@ defmodule Archethic.Contracts.Worker do
   defp schedule_trigger(_), do: :ok
 
   defp handle_new_transaction(next_transaction = %Transaction{}) do
-    [%Node{first_public_key: key} | _] =
-      next_transaction
-      |> Transaction.previous_address()
-      |> Election.chain_storage_nodes(P2P.authorized_and_available_nodes())
-
+    validation_nodes = P2P.authorized_and_available_nodes()
     # The first storage node of the contract initiate the sending of the new transaction
-    if key == Crypto.first_node_public_key() do
-      validation_nodes = P2P.authorized_and_available_nodes()
-
+    if trigger_node?(next_transaction) do
       P2P.broadcast_message(validation_nodes, %StartMining{
         transaction: next_transaction,
         validation_node_public_keys: Enum.map(validation_nodes, & &1.last_public_key),
         welcome_node_public_key: Crypto.last_node_public_key()
       })
+    else
+      DetectNodeResponsiveness.start_link(next_transaction.address, fn count ->
+        if trigger_node?(count) do
+          P2P.broadcast_message(validation_nodes, %StartMining{
+            transaction: next_transaction,
+            validation_node_public_keys: Enum.map(validation_nodes, & &1.last_public_key),
+            welcome_node_public_key: Crypto.last_node_public_key()
+          })
+        end
+      end)
     end
   end
 
+  defp trigger_node?(next_transaction = %Transaction{}, count \\ 0) do
+    %Node{first_public_key: key} =
+      next_transaction
+      |> Transaction.previous_address()
+      |> Election.chain_storage_nodes(P2P.authorized_and_available_nodes())
```

Comment:
you might reuse: `validation_nodes` here

Reply:
shall I pass validation nodes as arg to trigger_node? fn then ?